File: analyzer/web/models/BaseModel.js
class BaseModel{

    constructor(){}

    toJson(){
        var json={}
        var properties=Object.keys(this);
        for(var i=0;i<properties.length;i++){
            if(this[properties[i]] instanceof Figure){
                json[properties[i]]=this[properties[i]].toJson();
            }else{
                json[properties[i]]=this[properties[i]];
            }
        }
        return json;
    }
    
    getProperties(){
        return Object.keys(this);
    }
    
    fromJson(json){
        for(var key in json){
            this[key]=json[key];
        }
    }
}

File: analyzer/models/BaseModel.py
import inspect
class BaseModel:

    def __init__(self):
        pass
    
    def toJSON(self):
        properties=vars(self)
        if '_id' in properties.keys():
            del properties['_id']
        for key,value in properties.items():
            if inspect.isclass(value):
                properties[key]=value.toJSON()

        return properties
    
    def getProperties(self):
        return list(vars(self).keys())

    def loadFromDict(self,object_dict):
        for key,val in object_dict.items():
                setattr(self, key, val)